File: eBarberShop/UI/ebarbershop_mobile/lib/models/rezervacija/rezervacija_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija_insert_request.g.dart';

@JsonSerializable()
class RezervacijaInsertRequest{
  DateTime datum;
  DateTime vrijeme;
  bool? status;
  int korisnikId;
  int uposlenikId;
  int uslugaId;

  RezervacijaInsertRequest(this.datum, this.vrijeme, this.status, this.korisnikId, this.uposlenikId, this.uslugaId);

  factory RezervacijaInsertRequest.fromJson(Map<String, dynamic> json) => _$RezervacijaInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaInsertRequestToJson(this);
}

File: eBarberShop/eBarberShop.Model/Requests/RezervacijaInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace eBarberShop.Model.Requests
{
    public class RezervacijaInsertRequest
    {
        [Required(ErrorMessage = "Obavezno polje")]
        public DateTime Datum { get; set; }

        [Required(ErrorMessage = "Obavezno polje")]
        public DateTime Vrijeme { get; set; }

        public bool? Status { get; set; }

        [Required(ErrorMessage = "Obavezno polje")]
        public int KorisnikId { get; set; }

        [Required(ErrorMessage = "Obavezno polje")]
        public int UposlenikId { get; set; }

        [Required(ErrorMessage = "Obavezno polje")]
        public int UslugaId { get; set; }
    }
}
